src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/LegacyPropagator.cs

```diff
@@ -0,0 +1,190 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net;
+using System.Collections.Generic;
+
+namespace System.Diagnostics
+{
+    internal class LegacyPropagator : TextMapPropagator
+    {
+        internal static TextMapPropagator Instance { get; } = new LegacyPropagator();
+
+        public override IReadOnlyCollection<string> Fields { get; } = new HashSet<string>() { TraceParent, RequestId, TraceState, Baggage, CorrelationContext };
+
+        public override void Inject(Activity activity, object carrier, PropagatorSetterCallback setter)
+        {
+            if (activity is null || setter is null)
+            {
+                return;
+            }
+
+            string? id = activity.Id;
+            if (id is null)
+            {
+                return;
+            }
+
+            if (activity.IdFormat == ActivityIdFormat.W3C)
+            {
+                setter(carrier, TraceParent, id);
+                if (activity.TraceStateString is not null)
+                {
+                    setter(carrier, TraceState, activity.TraceStateString);
+                }
+            }
+            else
+            {
+                setter(carrier, RequestId, id);
+            }
+
+            InjectBaggage(carrier, activity.Baggage, setter);
+        }
+
+        public override void ExtractTraceIdAndState(object carrier, PropagatorGetterCallback getter, out string? traceId, out string? traceState)
+        {
+            if (getter is null)
+            {
+                traceId = null;
+                traceState = null;
+                return;
+            }
+
+            getter(carrier, TraceParent, out traceId, out _);
+            if (traceId is null)
+            {
+                getter(carrier, RequestId, out traceId, out _);
+            }
+
+            getter(carrier, TraceState, out traceState, out _);
+        }
+
+        public override IEnumerable<KeyValuePair<string, string?>>? ExtractBaggage(object carrier, PropagatorGetterCallback getter)
+        {
+            IEnumerable<KeyValuePair<string, string?>>? baggage = null;
```

Comment:
Nit: move this down, or invert the `if (getter is null)` to contain the rest of the code so that the last `return baggage` also handles the null case